### ISSUE TITLE
"Could not find handle.exe, please install from sysinternals" message floods the log

### ISSUE THREAD

**Issue by inikulshin** [rabbitmq/rabbitmq-management-agent#90]
It appears every ~5 seconds = ~17000 times = 1.7MB per day and is pretty useless. And also, makes analyzing logs harder.

It is not always possible to install **Sysinternals** on client's machine and, as you know, it can't be distributed with RabbitMQ (https://docs.microsoft.com/en-us/sysinternals/license-faq)

Is it possible to emit this error only **once**, for the first time only? And require to restart RabbitMQ after Sysinternals installation?

&ast; sometimes, without any explainable reason, we get "**Could not infer the number of file handles used: badarg**" instead.

**Comment by michaelklishin** [comment]
There is a 2000 year old Chinese saying that goes something like this: "a recurring issue logged only once will never be noticed". I find it to be on point.

I have changed log level to be `warning` for the case where `handle.exe` is missing. Having it available is really important as running out of file descriptors is one of the most commonly brought up operational issues.

I cannot comment on why the plugin fails to infer the number of file handles. It doesn't do anything particularly sophisticated to obtain this metric (although I wish there was a way to do it without running a subprocess/port command), you are welcome to take a look at the `rabbit_mgmt_external_stats` module (which is really small and mostly self-explanatory).
Ironically one of the reasons can be an OS limit that's been exceeded but only extensive monitoring can really tell.

**Comment by michaelklishin** [comment]
@dcorbacho @lukebakken let me know if you think we should switch to logging these only once.

**Comment by michaelklishin** [comment]
@inikulshin if you are aware of an alternative way of obtaining a process' file descriptor and socket count on Windows, let us know. We are all for dropping the `handle.exe` dependency for those who want to have decent monitoring. Can modern PowerShell do that? We are not Windows experts, unfortunately. But this is open source software so you can help in more than one way!

**Comment by lukebakken** [comment]
I'm re-adding code to only log the error once.

**Comment by Polylogic** [comment]
In addition to that, it would be nice to have an opt-out flag to disable this metric. On low-end pc the handle.exe eats CPU a lot, which is really annoying.

**Comment by lukebakken** [comment]
> In addition to that, it would be nice to have an opt-out flag to disable this metric

Don't install `handle.exe` and you'll be set.

**Comment by Polylogic** [comment]
My personal solution in this situation is to put compiled
```
// handle.exe
#include <stdio.h>
int main() { printf("  File : 123"); return 0; }
```
into AppData\Roaming\RabbitMQ directory, which is not great but allows to keep sysinternals suite installed on dev pc :)

**Comment by erichiller** [comment]
Same issue here,
You could run (in PowerShell)
```
( Get-Process -Id PIDHERE ).Handles
```

See example 8 in the [docs](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/get-process)
